packages/web3-core/src/web3_request_manager.ts

```diff
@@ -44,11 +44,11 @@ export class Web3RequestManager<
 	public constructor(provider?: SupportedProviders<API> | string | string, net?: Socket) {
 		super();
 
+		this._providers = Web3RequestManager.providers;
```

Comment:
```suggestion
		
```

Also remove :
`private readonly _providers: { [key: string]: Web3BaseProviderConstructor };`

and 

```
	public static get providers() {
		return {
			HttpProvider: HttpProvider as Web3BaseProviderConstructor,
			WebsocketProvider: WSProvider as Web3BaseProviderConstructor,
			IpcProvider: IpcProvider as Web3BaseProviderConstructor,
		};
	}
```
and directly return supported provider constructors inside get providers functions:

```
	public get providers() {
	return {
			HttpProvider: HttpProvider as Web3BaseProviderConstructor,
			WebsocketProvider: WSProvider as Web3BaseProviderConstructor,
			IpcProvider: IpcProvider as Web3BaseProviderConstructor,
		};
}
```

Reply:
Updated. Would be nice to review such points in the relevant PRs.